crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -212,13 +218,14 @@ impl Inner {
         );
 
         // surplus and fees calculation
-        let surplus = settlement.total_surplus(&external_prices);
+        let jit_order_uids = Postgres::get_missing_order_uids(
```

Comment:
I don't think this is the right approach, since it could lead to race conditions where an order that was inserted after the auction had been cut and sent to solvers still gets count. Doesn't the autopilot have the auction it sent to the solvers for a specific auction ID (we can read it from disk or cache it in memory). This auction contains all order uids that should count towards surplus (regular orders + smart orders which are allowed to produce surplus capturing JIT orders). Anything that's not in this list should be exempt from surplus.

Reply:
> since it could lead to race conditions where an order that was inserted after the auction had been cut and sent to solvers still gets count. Doesn't the autopilot have the auction it sent to the solvers for a specific auction ID (we can read it from disk or cache it in memory).

The concern about the race condition is true and there are even more false positives (e.g. solver can store and settle open orders which were not part of the auction due to missing native prices). However, AFAIK we don't have this information available so we settled on this intermediate solution which would already support CoW AMM orders while still being a strict improvement over the status quo even if it's not perfect, yet.
The goal was to unblock support for CoW AMM orders ASAP while allowing us to come up with a PR that implements the bullet proof solution with a DB migration.

Reply:
I am talking about storing the current auction that - just 10 seconds before - was sent to the solvers containing the problem they solved, somewhere potentially in memory (I am also pretty sure it's already stored in the db though. How else does the solution competition endpoint get the list of order uids it contained?)

Any issue with reindexing logic will likely not be fixed while the current auction is being solved (and even if we don't want it to apply to an auction that has already been sent to solvers).

With this approach we are adding more scattered DB reads (which are not using the same query logic as other places that try to reconstruct the auction) which I think creates more technical debt than is worth.

Reply:
I honestly think 3 is as good (if not less hacky) than this. If the format changes, we can also change the logic we use here (as this is only important for new auctions and never done on historic ones).

My main annoyance is that we are introducing yet another way of fetching data (which is doomed to be inconsistent). Refactoring this will be much harder to "prioritise" than adjusting the logic if we are changing the schema of the solver competition endpoint.

Reply:
I am ok with doing (3) or (4) as a temporary fix to unblock COWAMM but, after that, we should make a proper solution:
1. Save <auction, order_uid> into database (similar to how we store auction prices).
2. Implement logic of counting/not counting in the autopilot domain https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/domain/settlement/mod.rs. However, this is obviously still not ready to be used instead of "decoded_settlement.rs" (will work on it)

Reply:
Then let's go for option 3 to unblock this pr and get cow amms on the road.